config-ui/src/hooks/useWIndowSize.jsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react";
+
+function useWindowSize() {
+    const [windowSize, setWindowSize] = useState({
```

Comment:
Would it be better to initialize these to integer values rather than undefined, or is undefined desired? Maybe even a min-viewport size of 320x240.
```suggestion
    const [windowSize, setWindowSize] = useState({
      width: 320,
      height: 240,
    })
```